### PR TITLE
CSHARP-4550: Make MemberInitExpression work with struct also as long as suitable constructor exists.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

File: src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/MemberInitExpressionToAggregationExpressionTranslator.cs [hunk 1]
```diff
@@ -29,42 +29,51 @@ internal static class MemberInitExpressionToAggregationExpressionTranslator
...
                 memberMap.SetSerializer(valueTranslation.Serializer);
+                membersBeingDeserialized.Add(memberMap);
+                computedFields.Add(AstExpression.ComputedField(memberMap.ElementName, valueTranslation.Ast));
             }
 
+            //if (expression.Type.IsValueType)
+            //{
+            //    EnsureConstructorRequiredToDeserializeStructIsMapped(expression, classMap, membersBeingDeserialized);
+            //}
+
```
Comment (rstam) [inline src/MongoDB.Driver/Linq/Linq3Implementation/Translators/ExpressionToAggregationExpressionTranslators/MemberInitExpressionToAggregationExpressionTranslator.cs:76]: I haven't phyically removed the `EnsureConstructorRequiredToDeserializeStructIsMapped ` method (yet).

I just don't call it to see what happens.

**Review by sanych-sun — APPROVED** [review]
LGTM

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Review by sanych-sun — APPROVED** [review]
LGTM

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**